packages/config/src/handlers/application/applicationConfigHandler.ts

```diff
@@ -132,15 +132,22 @@ export async function updateApplicationConfigHandler(
   h: Hapi.ResponseToolkit
 ) {
   try {
-    const currentConfig = await getApplicationConfig()
+    let applicationConfig
+    const configFromDB = await ApplicationConfig.findOne({})
     const changeConfig = request.payload as IApplicationConfigurationModel
-    const applicationConfig = merge(currentConfig, changeConfig)
+
+    if (configFromDB !== null) {
+      applicationConfig = merge(configFromDB, changeConfig)
+    }
+    applicationConfig = changeConfig
     await ApplicationConfig.findOneAndUpdate(
       {},
       { $set: applicationConfig },
       { upsert: true }
     )
-    return h.response(applicationConfig).code(201)
+    const currentConfig = await getApplicationConfig()
+
+    return h.response(merge(currentConfig, applicationConfig)).code(201)
```

Comment:
@kayumuzzaman Could we not just return `getApplicationConfig` as that already merges the values gotten from country-config with the ones from db?